proxy-recorder/process_gatling_recording.rb

```diff
@@ -20,17 +22,171 @@ def lookup_request_match(line, prefix)
+          return File.absolute_path(file)
+        end
+      end
+    end
+  end
+  nil
+end
+
+def prompt(msg)
+  $stdout.write msg
+  result = $stdin.gets.strip
+  if result.empty?
+    nil
+  else
+    result
+  end
+end
+
 def comment_line(line)
   "// #{line}"
 end
 
+def get_simulation_runner_root_dir()
+  File.join(File.dirname(__FILE__), "..", "simulation-runner")
+end
+
+def validate_infile_path(infile)
+  infile_dir = File.absolute_path(File.dirname(infile))
+  sim_dir = File.absolute_path(
+                    File.join(get_simulation_runner_root_dir(),
+                              "src", "main", "scala", "com", "puppetlabs",
+                              "gatling", "node_simulations"))
+
+  unless infile_dir == sim_dir
+    puts "ERROR: Simulation recording must be placed in this directory: '#{sim_dir}'"
+    puts "(specified input file is in dir '#{infile_dir}')"
+    exit 1
+  end
+end
+
+def find_certname(text)
+  matches = text.match(/\n\s*\.get\("\/puppet\/v3\/node\/([^\?]+)\?environment/)
+  unless matches
+    puts "Unable to find certname (from node request) in recording!"
+    exit 1
+  end
+
+  certname = matches[1]
+
+  puts "Found certname: '#{certname}'"
+  puts
+
+  certname
+end
+
+def find_report_request_info(text)
+  matches = text.match(/\n\s*\.put\("\/puppet\/v3\/report[^"]+"\)\s*\n\s*\.headers\(([^\)]+)\)\s*\n\s*\.body\(RawFileBody\("([^"]+)"\)\)\)\s*\n/)
+  unless matches
+    puts "Unable to find report request in recording!"
+    exit 1
+  end
+
+  puts "Found report request."
+
+  result = {:request_headers_varname => matches[1],
+            :request_txt_file => matches[2]}
+
+  request_txt = File.absolute_path(
+                        File.join(get_simulation_runner_root_dir(), "user-files", "bodies",
+                                  result[:request_txt_file]))
+
+  unless File.file?(request_txt)
+    puts "Unable to find report request body file!  Expected to find it at #{request_txt}"
+    exit 1
+  end
+
+  result[:request_txt_file_path] = request_txt
+
+  puts "\tReport request headers var: #{result[:request_headers_varname]}"
+  puts "\tReport request text file: #{result[:request_txt_file_path]}"
+  puts
+  result
+end
+
+def find_node_config(simulation_class)
+  node_configs_dir = File.join(get_simulation_runner_root_dir(), "config", "nodes")
+  existing_node_config = grep_dir(node_configs_dir, /"simulation_class"\s*:\s*"#{simulation_class}"/)
+  existing_node_config
+end
+
+def generate_node_config(simulation_class, certname)
+  node_configs_dir = File.join(get_simulation_runner_root_dir(), "config", "nodes")
+  node_config_filename = simulation_class.sub(/.*\.([^\.]+)$/, '\1') + ".json"
+  node_config_file = File.absolute_path(File.join(node_configs_dir, node_config_filename))
+  if File.exists?(node_config_file)
+    puts "ERROR: file already exists at path '#{node_config_file}'; exiting."
+    exit 1
+  end
+
+  data = prompt("Node config file name? [#{node_config_filename}]> ")
```

Comment:
Not a big deal but it might be nice to have a way to specify these as command line arguments to the tool and, if specified, use those and avoid prompting.  I generally like command line args over prompting because it's easier to repeat the command later from shell history, etc.